### PR TITLE
React redux store and more user/project servlets

### PR DESCRIPTION
**Pull request by michaellawes**

### CODE REVIEW

File: src/main/webapp/components/Auth/Auth.js [hunk 1]
```diff
@@ -196,8 +217,10 @@ const Auth = ({ history }) => {
       }
       if (usernameExists || emailExists) return;
 
-      const signInInfo = { username: username, email: email };
-      //dispatch(signIn(signInInfo));
+      dispatch(signIn({ username: username, email: email, name: name }));
+      dispatch(loadUsers(username));
+      dispatch(loadProjects(username));
+      console.log(user.username);
```
Comment (sarahrittgers26) [inline src/main/webapp/components/Auth/Auth.js]: please remove console.log statements, assuming they're used for debugging

File: src/main/webapp/components/Api/Api.js [hunk 2]
```diff
@@ -1,6 +1,6 @@
 import axios from 'axios';
-//const baseUrl = 'http://localhost:8080'
-const baseUrl = 'http://spring21-sps-18.appspot.com'
+const baseUrl = 'http://localhost:8080';
+//const baseUrl = 'http://spring21-sps-18.appspot.com';
```
Comment (sarahrittgers26) [inline src/main/webapp/components/Api/Api.js:4]: comment out the localhost url?

File: src/main/webapp/actions/index.js [hunk 3]
```diff
@@ -1,11 +1,131 @@
...
+      // if selector is true then only add online projects otherwise add offline
+      if (current.bothActive == selector) {
+        let project = { collaborator: current.partner, projectid: current.projectid,
+	        title: current.title, html: current.html, css: current.css, 
+	        js: current.js };
+	selection.push(project);
+      }
+    }
+    return selection;
+}
```
Comment (sarahrittgers26) [inline src/main/webapp/actions/index.js]: if this doesn't need to be exported, let's move it above all the exports. same for userSelector

File: src/main/java/com/google/sps/servlets/GetUsersServlet.java [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+		 // Return project objects to frontend
+		 response.setContentType("application/json");
+		 response.getWriter().println(gson.toJson(users));
+	}
+
+	// Get users associated with user from Datastore
+	private FormattedUser[] loadUsers(String username, Datastore datastore) 
+			throws DatastoreException {
+		 // Initialize Array list for usernames of project partners	
+		 List<String> allUsers = new ArrayList<>();
```
Comment (sarahrittgers26) [inline src/main/java/com/google/sps/servlets/GetUsersServlet.java]: Why not just initialize this as a set and add the users to it directly? I don't see allUsers being used anywhere else besides being copied to the set

File: src/main/java/com/google/sps/servlets/GetUsersServlet.java [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+			// Determine whether partner is stored as user1 or user2
+			String partner = user1.equals(username) ? user2 : user1;
+			collaborators.add(partner);
+		 }
+		 return collaborators;
+	}
+
+	// Check if user is active
+	private boolean userIsActive(String lastLogin) {
+		// Convert to LocalDateTime and check if within 1 minute
```
Comment (sarahrittgers26) [inline src/main/java/com/google/sps/servlets/GetUsersServlet.java:118]: Please explain why we need to check the login time is within 1 minute

File: src/main/webapp/actions/index.js [hunk 6]
```diff
@@ -1,11 +1,131 @@
...
+  }
+};
+
+// Update active users
+export const updateActiveUsers = (userList) => {
+  return {
+    type: ACTION.UPDATE_ACTIVE_USERS,
+    payload: userList
+  }
+}
```
Comment (sarahrittgers26) [inline src/main/webapp/actions/index.js]: Please add semicolons at the end, here and below

File: src/main/webapp/actions/index.js [hunk 7]
```diff
@@ -1,11 +1,131 @@
...
+    const response = await axios.get(url);
+    dispatch({ type: ACTION.UPDATE_ONLINE_PROJECTS, payload: projectSelector(true, response.data) });
+    dispatch({ type: ACTION.UPDATE_OFFLINE_PROJECTS, payload: projectSelector(false, response.data) });
+  }
+}
+
+// Return either online or offline projects
+const projectSelector = (selector, projects) => {
+    const selection = [];
+    for (var i = 0;i < projects.length; i++) {
```
Comment (sarahrittgers26) [inline src/main/webapp/actions/index.js]: nit: add a space after the semicolon

`for (var i = 0; i < projects.length; i++) {`

**Review by sarahrittgers26 — APPROVED** [review]

File: src/main/webapp/components/Auth/Auth.js [hunk 8]
```diff
@@ -223,16 +246,19 @@ const Auth = ({ history }) => {
...
 	  setPasswordError(true);
 	  setPasswordErrorMsg('Username or password entered incorrectly');
 	} else {
-	  const signInInfo = { username: username, email: errorAndEmail[1] };
-	  //dispatch(signIn(signInInfo));
+	  dispatch(signIn({ username: username, email: errorAndInfo[1], 
+		  name: errorAndInfo[2] }));
+	  dispatch(loadUsers(username));
+	  dispatch(loadProjects(username));
+	  console.log(user.username);
```
Comment (MufaroMakiwa) [inline src/main/webapp/components/Auth/Auth.js]: You can as well remove this console log statement

**Review by MufaroMakiwa — APPROVED** [review]

**Pull request merged**